alerts/cluster-monitoring-operator/NodeFileDescriptorLimit.md

```diff
@@ -0,0 +1,30 @@
+# NodeFileDescriptorLimit
+
+## Meaning
+
+This alert is triggered when a node's kernel is found to be running out of
+available file descriptors -- a `warning` level alert at greater than 70% usage
+and a `critical` level alert at greater than 90% usage.
+
+## Impact
+
+Applications on the node may no longer be able to open and operate on
+files. This is likely to have severe consequences for anything scheduled on this
+node.
+
+## Diagnosis
+
+You can open a shell on the node and use the standard Linux utilities to
+diagnose the issue:
+
+```console
+$ NODE_NAME='<value of instance label from alert>'
+
+$ oc debug "node/$NODE_NAME"
+$ lsof -n
```

Comment:
Do we maybe want to add `ulimit` to also print out the limit?

Reply:
I actually don't think this one has anything to do with ulimit, this is about the overall limits in the kernel, so we could add something like: `# sysctl -a | grep 'fs.file'`

As far as I know, the kernel limit is based on amount of memory, and given the size of most OpenShift nodes, I would be super surprised if this ever fires honestly.

Reply:
Added sample output from the `sysctl` command on an OpenShift 4.9 master node.